activerecord/lib/active_record/persistence.rb

```diff
@@ -332,16 +332,18 @@ def increment(attribute, by = 1)
     # Saving is not subjected to validation checks. Returns +true+ if the
     # record could be saved.
     def increment!(attribute, by = 1)
-      increment(attribute, by).update_attribute(attribute, self[attribute])
+      increment(attribute, by)
+      change = public_send(attribute) - (attribute_was(attribute.to_s) || 0)
+      self.class.update_counters(id, attribute => change)
+      clear_attribute_change(attribute) # eww
+      self
```

Comment:
I think this changes the behavior from the documentation, which says that these methods return `true` on success, but now it seems to always return `self`?